core/chaincode/platforms/golang/hash.go

```diff
@@ -138,7 +135,11 @@ func getCodeFromHTTP(path string) (codegopath string, err error) {
 	//     . more secure
 	//     . as we are not downloading OBC, private, password-protected OBC repo's become non-issue
 
-	env[gopathenvIndex] = "GOPATH=" + codegopath + ":" + origgopath
+	os.Setenv("GOPATH", codegopath+":"+origgopath)
+	// Get a copy of that new env for the go get command
+	env := os.Environ()
+	// and reset GOPATH to its original value
+	os.Setenv("GOPATH", origgopath)
```

Comment:
@lehors the original code in old line 141 is correct. We don't want to override the systems GOPATH with the new Env (new lines 138-142). We want to just change the "GOPATH" element in the original "env" var locally to be passed to exec.Command.